src/ImageSharp.Drawing/Processing/FillRectangleExtensions.cs

```diff
@@ -61,5 +62,55 @@ public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessing
+        /// <param name="rectangle">The rectangle shape.</param>
+        /// <returns>The <see cref="Image{TPixel}"/>.</returns>
+        public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessingContext<TPixel> source, TPixel color, Rectangle rectangle)
+            where TPixel : struct, IPixel<TPixel>
+            => source.Fill(GraphicsOptions.Default, new SolidBrush<TPixel>(color), rectangle);
+
+        /// <summary>
+        /// Fills the image in the shape of the provided rectangle with the specified brush.
+        /// </summary>
+        /// <typeparam name="TPixel">The type of the color.</typeparam>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="brush">The details how to fill the region of interest.</param>
+        /// <param name="rectangle">The rectangle shape.</param>
+        /// <returns>The <see cref="Image{TPixel}"/>.</returns>
+        public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessingContext<TPixel> source, IBrush<TPixel> brush, Rectangle rectangle)
+            where TPixel : struct, IPixel<TPixel>
+            => source.Fill(GraphicsOptions.Default, brush, rectangle);
+
+        /// <summary>
+        /// Fills the image in the shape of the provided rectangle with the specified color.
+        /// </summary>
+        /// <typeparam name="TPixel">The type of the color.</typeparam>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="options">The graphics options.</param>
+        /// <param name="color">The color.</param>
+        /// <param name="rectangle">The rectangle shape.</param>
+        /// <returns>The <see cref="Image{TPixel}"/>.</returns>
+        public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessingContext<TPixel> source, GraphicsOptions options, TPixel color, Rectangle rectangle)
+            where TPixel : struct, IPixel<TPixel>
+            => source.Fill(options, new SolidBrush<TPixel>(color), rectangle);
+
+        /// <summary>
+        /// Fills the image in the shape of the provided rectangle with the specified brush.
+        /// </summary>
+        /// <typeparam name="TPixel">The type of the color.</typeparam>
+        /// <param name="source">The image this method extends.</param>
+        /// <param name="options">The graphics options.</param>
+        /// <param name="brush">The details how to fill the region of interest.</param>
+        /// <param name="rectangle">The rectangle shape.</param>
+        /// <returns>The <see cref="Image{TPixel}"/>.</returns>
+        public static IImageProcessingContext<TPixel> Fill<TPixel>(this IImageProcessingContext<TPixel> source, GraphicsOptions options, IBrush<TPixel> brush, Rectangle rectangle)
+            where TPixel : struct, IPixel<TPixel>
+            => source.ApplyProcessor(new FillProcessor<TPixel>(brush, options), rectangle);
```

Comment:
If I understand @tocsoft 's comment well, it would be the best to select between `FillRegionProcessor` and `FillProcessor` at this point, depending on the value of `options.Antialias`. All overloads should delegate into this method, including the ones which haven't been touched.

Reply:
Been thinking about this, and now I'm unsure about the expected default behavior for these use cases.
- If a rectangle is being filled with a solid color, is it expected to antialias corners by default?
- What about non-solid brushes?

Reply:
I thought the internal processor code checked to see if the bush was a solid color?